codewiki/src/be/cluster_modules.py

```
@@ -47,41 +50,53 @@ def cluster_modules(
-            logger.error(f"Invalid LLM response format - missing component tags: {response[:200]}...")
+            logger.error(
+                f"Invalid LLM response format - missing component tags: {response[:200]}..."
+            )
             return {}
-        
-        response_content = response.split("<GROUPED_COMPONENTS>")[1].split("</GROUPED_COMPONENTS>")[0]
+
+        response_content = response.split("<GROUPED_COMPONENTS>")[1].split("</GROUPED_COMPONENTS>")[
+            0
+        ]
         module_tree = eval(response_content)
```

Ruff (0.14.10)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)